org.eclipse.jdt.core.tests.model/src/org/eclipse/jdt/core/tests/model/JavaSearchBugsTests.java

```diff
@@ -10737,11 +10737,13 @@ public void run() {
 		    }
 		};
 		Thread thread = new Thread(search);
+		long start = System.currentTimeMillis();
```

Comment:
please use nanotime, as currentTimeMillis is not guaranteed to be monotonic. see https://github.com/eclipse-jdt/eclipse.jdt.core/pull/3383/files you coudl use the` long timeoutNanos = System.nanoTime() + this.timeout * 1_000_000L` pattern here too